### PR TITLE
Engine clean up

### PR DESCRIPTION
**Pull request by saig0**
## Description

* remove old BPMN handlers
* remove BPMN steps and lifecycle binding
* remove step context, event output and workflow engine state  

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by saig0** [comment]
Some classes and packages should be moved. I'll open a separate issue/PR for this.

**Review by korthout — CHANGES_REQUESTED** [review]
Wow, this took some effort to review and I can imagine that the same goes for writing it. Great work! I have some remarks here and there, but nothing major. This work really cleans up the engine!

**Comment by korthout** [comment]
I also wondered whether you wanted to move the classes around the packages and rename some packages in this PR like we discussed.

**Comment by saig0** [comment]
@korthout, I applied your hints. Please have a second look.

I want to move/rename the packages in a follow-up PR.

File: engine/src/test/java/io/zeebe/engine/processor/workflow/ExpressionProcessorCorrelationKeyResultHandlerTest.java [hunk 1]
```diff
@@ -23,9 +23,11 @@
 @RunWith(MockitoJUnitRunner.class)
 public class ExpressionProcessorCorrelationKeyResultHandlerTest {
 
-  public static final String TEST_EXPRESSION = "expression";
+  private static final String TEST_EXPRESSION = "expression";
+  private static final long VARIABLE_SCOPE_KEY = 21;
```
Comment (korthout) [inline engine/src/test/java/io/zeebe/engine/processor/workflow/ExpressionProcessorCorrelationKeyResultHandlerTest.java:27]: I'm still not really sure about this. The name implies something about the value, but it still seems like a magic number. Why is it 21?

**Review by korthout — APPROVED** [review]
LGTM, one last question, but great work so here's the approval. 👍

File: engine/src/test/java/io/zeebe/engine/processor/workflow/ExpressionProcessorCorrelationKeyResultHandlerTest.java (hunk 1 above)
Comment (saig0) [reply engine/src/test/java/io/zeebe/engine/processor/workflow/ExpressionProcessorCorrelationKeyResultHandlerTest.java:27]: The variable scope key is not used in the test but it needs to be passed into the context. So, I chose a random number :)

**Comment by saig0** [comment]
bors retry

**Pull request merged**